File: tests/data/program_analysis/strings/str02.f
C     File: str02.f
C     This program tests declaration and initialization of strings.

      program main
      character(len = 10) :: str1 = "abcdef"

      write (*, 10) "str1", len(str1), str1

 10   format(A, ': len = ', I2, '; value = "', A, '"')

      stop
      end program main

File: tests/data/program_analysis/strings/Translation/str02.py
from delphi.translators.for2py.format import *
from delphi.translators.for2py.strings import *

def main():
    str1 = String(10, "abcdef")

    fmt_10 = Format(['A', '": len = "', 'I2', '"; value = \""', 'A', '"\""'])
    write_str = fmt_10.write_line(["str1", len(str1), str1])
    sys.stdout.write(write_str)

main()
